src/scores/continuous/interval_impl.py

```diff
@@ -0,0 +1,100 @@
+
+
+def quantile_interval_score(  # pylint: disable=R0914
+    fcst_lower_qtile: xr.DataArray,
+    fcst_upper_qtile: xr.DataArray,
+    obs: xr.DataArray,
+    lower_qtile_level: float,
+    upper_qtile_level: float,
+    *,  # Force keywords arguments to be keyword-only
+    reduce_dims: Optional[FlexibleDimensionTypes] = None,
+    preserve_dims: Optional[FlexibleDimensionTypes] = None,
+    weights: Optional[xr.DataArray] = None,
+) -> xr.Dataset:
+    """
+    Calculates the qauntile interval score for interval forecasts. This score penalizes
+    the interval's width and whether the observation value lies outside the interval.
+
+    Args:
+        fcst_lower_qtile: array of forecast values at the lower quantile.
+        fcst_upper_qtile: array of forecast values at the upper quantile.
+        obs: array of observations
+        lower_qtile_level: Quantile level between 0 and 1 (exclusive) for `fcst_lower_qtile`
+        upper_qtile_level: Quantile level between 0 and 1 (exclusive) for `fcst_upper_qtile`
+        reduce_dims: Optionally specify which dimensions to reduce when
+            calculating the quantile interval score. All other dimensions will be preserved.
+            As a special case, 'all' will allow all dimensions to be reduced. Only one
+            of `reduce_dims` and `preserve_dims` can be supplied. The default behaviour
+            if neither are supplied is to reduce all dims.
+        preserve_dims: Optionally specify which dimensions to preserve when calculating
+            quantile interval score. All other dimensions will be reduced. As a special case,
+            'all' will allow all dimensions to be preserved. In this case, the result will be in
+            the same shape/dimensionality as the forecast, and the errors will be the quantile
+            score at each point (i.e. single-value comparison against observed), and the
+            forecast and observed dimensions must match precisely. Only one of `reduce_dims`
+            and `preserve_dims` can be supplied. The default behaviour if neither are supplied
+            is to reduce all dims.
+        weights: Optionally provide an array for weighted averaging (e.g. by area, by latitude,
+            by population, custom)
+
+    Returns:
+        A Dataset with the dimensions specified in `dims`.
+        The dataset will have the following data variables:
+
+        - interval_width_penalty: fcst_upper_qtile - fcst_lower_qtile
+        - overprediction_penalty: ((1/lower_qtile_level)*(fcst_lower_qtile-obs)).clip(min=0)
+        - underprediction_penalty: ((1/upper_qtile_level)*(fcst_upper_qtile-obs)).clip(min=0)
+        - total: sum of all penalties
+
+    Raises:
+        ValueError: If not (0 < lower_qtile_level < upper_qtile_level < 1).
+        ValueError: If (fcst_lower_qtile > fcst_upper_qtile).any().
+
+    References:
+        T. Gneiting, and Raftery A.E., "Strictly proper scoring rules, prediction and
```

Comment:
For references, we have been moving towards using the APA 7 style guide, see: https://apastyle.apa.org/style-grammar-guidelines/references/examples/journal-article-references

Reply:
It has now been fixed